shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -324,13 +327,14 @@ private void applyStateToSelection(TextInputChannel.TextEditState state) {
     @SuppressWarnings("deprecation")
     private boolean isRestartAlwaysRequired() {
         InputMethodSubtype subtype = mImm.getCurrentInputMethodSubtype();
-        if (subtype == null) {
+        // Impacted devices all shipped with Android Lollipop or newer.
+        if (subtype == null || Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
             return false;
         }
-        String language = (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N)
-                ? subtype.getLanguageTag()
-                : subtype.getLocale();
-        return Build.MANUFACTURER.equals("samsung") && language.equals("ko");
+        String keyboardName = Settings.Secure.getString(mView.getContext().getContentResolver(), Settings.Secure.DEFAULT_INPUT_METHOD);
```

Comment:
Does this API call read from disk? I thought this API was slow because it ended up doing I/O, but I could be mistaken here.

Reply:
Let me look it up

Reply:
There seems to be no mention of it storing it on disk, which I would imagine if it were stored on disk, would at least be mentioned a few times.

Reply:
It reads from something called sNameValueCache, which I am assuming reads from disk at some point but the values should be cached when we use it here.

Reply:
In any case, I added short circuiting to prevent non-samsung devices from performing the lookup.

Reply:
Thanks for looking into it, this LGTM.